eth/core/precompile/method_validation.go

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// Copyright (c) 2023 Berachain Foundation
+//
+// Permission is hereby granted, free of charge, to any person
+// obtaining a copy of this software and associated documentation
+// files (the "Software"), to deal in the Software without
+// restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following
+// conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+// OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+
+package precompile
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+)
+
+// ValidateBasic checks if the precompile method argument and return types match
+// the abi's argument and return types. This is for overloaded Solidity functions and general
+// validation.
+func (m *Method) ValidateBasic() error {
+	implMethod := m.execute
+	abiMethod := m.abiMethod
+
+	if len(m.abiMethod.Outputs) == 0 {
+		// the Solidity compiler requires that precompiles must return at least one value.
+		// see https://github.com/berachain/polaris/issues/491 for more information.
+
+		//nolint:lll // error message.
+		panic("The Solidity compiler requires all precompile functions to return at least one value. Consider returning a boolean.")
+	}
+
+	// first two args of Go precompile implementation are the receiver contract and the
+	// context, so we skip those.
+	if implMethod.Type.NumIn()-2 != len(abiMethod.Inputs) {
+		return errors.New("number of arguments mismatch")
+	}
+
+	// last parameter of Go precompile implementation is an error (for reverts),
```

Comment:
This is a huge nit, and something that we should discuss more broadly with @calbera as well. But generally I lean towards proper capitalization and sentence structure for comments. 

I’m good either way, but we decide and then be consistent throughout the repo. 

1) Either proper sentence structure and capitalization (my preference) “SENTENCES”

2) start all comments with lowercase and no periods. “NOT SENTENCES”

Reply:
does the linter check for this? think sentences are much cleaner and make the repo look nicer

Reply:
I would rather avoid over-commenting and only commenting to explain some logic/tradeoff that isnt explicitly made clear by the code. hence most of the comments before a function explain those decisions and effects. Inside the function it should be just enough comments to allow the reader to follow the logic and sentence comments feel a lot "heavier"